app/helpers/application_helper.rb

```diff
@@ -527,4 +527,12 @@ def password_complexity_requirements
     s += raw '<br /><em>' + rules + '</em>' unless rules.empty?
     s
   end
+
+  def settings_project_path(project, opts = {})
+    if opts[:tab]
+      url_for controller: "project_settings/#{opts[:tab]}", action: 'show', only_path: true, id: project
+    else
+      url_for controller: "project_settings/generic", action: 'show', only_path: true, id: project
```

Comment:
I am not sure if you actually need this helper method but if you do, this can be shortened to 
```
send("settings_#{opts[:tab] || 'generic'}_project_path", project)
```